src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+
+        public CSharpAddBracesDiagnosticAnalyzer()
+        {
+            _noneAddBracesDiagnosticDescriptor = CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity.Hidden);
+            _infoAddBracesDiagnosticDescriptor = CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity.Info);
+            _warningAddBracesDiagnosticDescriptor = CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity.Warning);
+            _errorAddBracesDiagnosticDescriptor = CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity.Error);
+
+            _severityToAddBracesDescriptorMap =
+                new Dictionary<DiagnosticSeverity, DiagnosticDescriptor>
+                {
+                    {DiagnosticSeverity.Hidden, _noneAddBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Info, _infoAddBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Warning, _warningAddBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Error, _errorAddBracesDiagnosticDescriptor },
+                };
+
+            _noneRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Hidden);
+            _infoRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Info);
+            _warningRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Warning);
+            _errorRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Error);
+
+            _severityToRemoveBracesDescriptorMap =
+                new Dictionary<DiagnosticSeverity, DiagnosticDescriptor>
+                {
+                    {DiagnosticSeverity.Hidden, _noneRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Info, _infoRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Warning, _warningRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Error, _errorRemoveBracesDiagnosticDescriptor },
+                };
+        }
+
+        private DiagnosticDescriptor CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.AddBracesDiagnosticId,
+                s_addBracesLocalizableTitle,
+                s_addBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+
+        private DiagnosticDescriptor CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.RemoveBracesDiagnosticId,
+                s_removeBracesLocalizableTitle,
+                s_removeBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+        public bool RunInProcess => false;
+
```

Comment:
I dont understand `RunInProcess` very well. But I think this needs to be true, because you're now reading options (because they are not available in the devhub process, atleast currently)? @heejaechang or @dpoeschl can clarify this.